### PR TITLE
Data product standards for 'aqicn' and 'prh' productizers

### PR DESCRIPTION
**Pull request by samuelashraff**

### CODE REVIEW

File: draft/DataProducts/AirQuality/Current/Info.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            "type": "number",
+            "example": 30
+          },
+          "timestamp": {
+            "title": "Current timestamp in RFC 3339 format",
+            "type": "string",
+            "example": "2020-04-03T13:00:00Z"
+          },
+          "attribution": {
+            "title": "Source Attribution",
```
Comment (lietu) [inline draft/DataProducts/AirQuality/Current/Info.json:98]: It's good to say "required to be shown where this value is used"

File: draft/DataProducts/AirQuality/Current/Info.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            "type": "array",
+            "example": "[Moscow State environmental monitoring (Государственный экологический мониторинг в Москве)World Air Quality Index Project"],
```
Comment (lietu) [inline draft/DataProducts/AirQuality/Current/Info.json:100]: This is somehow broken - the `]` is outside the `"` and even still this value looks strange?

File: draft/DataProducts/AirQuality/Current/Info.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          },
+          "type": {
+            "title": "Error Type",
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (lietu) [inline draft/DataProducts/AirQuality/Current/Info.json:158]: Add EOL

File: draft/DataProducts/AirQuality/Current/Info.jsonld [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "@context": {
+    "@version": 1.1,
+    "saref": {
+      "@id": "https://w3id.org/saref#",
+      "@prefix": true
+    },
+    "airquality": {
+      "@id": "https://www.environment.nsw.gov.au/topics/air/understanding-air-quality-data/glossary-of-air-quality-terms#aqi",
```
Comment (lietu) [inline draft/DataProducts/AirQuality/Current/Info.jsonld:9]: In the Wiki ensure we have a mention of this being used, but being a low quality source

File: draft/DataProducts/Company/BasicInfo.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
+{
+  "openapi": "3.0.2",
+  "info": {
+    "title": "Company Information",
+    "description": "Data Product for company information given by the business id parameter",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:5]: "given by the business id parameter" is useless repetition of information already available

File: draft/DataProducts/Company/BasicInfo.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/Company/BasicInfo": {
+      "post": {
+        "summary": "/Company Data Product",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:11]: What is this `/`? Also this is not a "Company" data product

File: draft/DataProducts/Company/BasicInfo.json [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        "description": "Company Information given a certain business id parameter",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:12]: Same as above

File: draft/DataProducts/Company/BasicInfo.json [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        "operationId": "company_data_product_post",
+        "parameters": [],
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/PrhRequest"
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:19]: How is PRH related?

Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.json:19]: You will likely need to rename the class in code to get it to use a better name here

File: draft/DataProducts/Company/BasicInfo.json [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/PrhResponse"
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:31]: Same

File: draft/DataProducts/Company/BasicInfo.json [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    "schemas": {
+      "PrhRequest": {
+        "title": "PrhRequest",
+        "required": [
+          "company_id"
+        ],
+        "type": "object",
+        "properties": {
+          "company_id": {
+            "title": "CompanyID",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:60]: This should likely be human readable and thus include spaces

File: draft/DataProducts/Company/BasicInfo.json [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+          "business_id",
+          "company_form",
+          "registration_date"
+        ],
+        "type": "object",
+        "properties": {
+          "name": {
+            "title": "Name of the company",
+            "type": "string",
+            "example": "Digitalliving"
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:79]: Bad example considering it's not a real value

File: draft/DataProducts/Company/BasicInfo.json [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+          },
+          "type": {
+            "title": "Error Type",
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:152]: Add EOL

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    "vocab": {
+      "@id": "https://vocabulary.lifeengine.io/Business/#",
+      "@prefix": true
+    },
+    "schema": {
+      "@id": "https://schema.org/LocalBusiness",
+      "@prefix": true
+    },
+    "name": "rov:legalName",
+    "registrationDate": "vocab:RegistrationDate",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:17]: There is no field `registrationDate` in the response

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+    "business_id": "schema:leiCode",
+    "company_form": "rov:orgType"
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:19]: https://www.w3.org/TR/vocab-regorg/#orgType does not work

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+      "@id": "https://vocabulary.lifeengine.io/Business/#",
+      "@prefix": true
+    },
+    "schema": {
+      "@id": "https://schema.org/LocalBusiness",
+      "@prefix": true
+    },
+    "name": "rov:legalName",
+    "registrationDate": "vocab:RegistrationDate",
+    "business_id": "schema:leiCode",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:18]: This is not a LEI code

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    "vocab": {
+      "@id": "https://vocabulary.lifeengine.io/Business/#",
+      "@prefix": true
+    },
+    "schema": {
+      "@id": "https://schema.org/LocalBusiness",
+      "@prefix": true
+    },
+    "name": "rov:legalName",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:16]: https://www.w3.org/TR/vocab-regorg/#legalName does not work

File: draft/DataProducts/Company/BasicInfo.jsonld (hunk 15 above)
Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.jsonld:18]: also 
image

**Comment by lietu** [comment]
Pretty good first try, found a few things to correct.

File: draft/DataProducts/Company/BasicInfo.json (hunk 8 above)
Comment (samuelashraff) [reply draft/DataProducts/Company/BasicInfo.json:19]: Rename 'PrhRequest' to what?
Should I also rename the 'PrhResponse' class?

Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.json:19]: It's a good idea to align with the data product name, e.g. `CompanyBasicInfoRequest` & `...Response`

File: draft/DataProducts/AirQuality/Current/Info.json (hunk 1 above)
Comment (samuelashraff) [reply draft/DataProducts/AirQuality/Current/Info.json:98]: So I'll add a "description" to it?

File: draft/DataProducts/Company/BasicInfo.json (hunk 11 above)
Comment (samuelashraff) [reply draft/DataProducts/Company/BasicInfo.json:79]: Digital Living International Oy
Better?

Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.json:79]: If it is a real value from the registry, yes

File: draft/DataProducts/AirQuality/Current/Info.json (hunk 1 above)
Comment (lietu) [reply draft/DataProducts/AirQuality/Current/Info.json:98]: That might be one way to do that

File: draft/DataProducts/AirQuality/Current/Info.jsonld (hunk 4 above)
Comment (samuelashraff) [reply draft/DataProducts/AirQuality/Current/Info.jsonld:9]: What wiki?

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 17]
```diff
@@ -10,12 +10,12 @@
...
+      "@id": "https://schema.org/",
       "@prefix": true
     },
-    "name": "rov:legalName",
-    "registrationDate": "vocab:RegistrationDate",
+    "name": "rov:ref_rov_legalname",
+    "registration_date": "vocab:RegistrationDate",
     "business_id": "schema:leiCode",
-    "company_form": "rov:orgType"
+    "company_form": "rov:ref_rov_orgType"
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:19]: You should move the `ref_rov_` to the prefix for clarity here

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 18]
```diff
@@ -10,12 +10,12 @@
       "@prefix": true
     },
     "schema": {
-      "@id": "https://schema.org/LocalBusiness",
+      "@id": "https://schema.org/",
       "@prefix": true
     },
-    "name": "rov:legalName",
-    "registrationDate": "vocab:RegistrationDate",
+    "name": "rov:ref_rov_legalname",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:16]: You should move the ref_rov_ to the prefix for clarity here

File: draft/DataProducts/Company/BasicInfo.json (hunk 7 above)
Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.json:12]: Still says `given a certain business id`

File: draft/DataProducts/Company/BasicInfo.jsonld [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+    "vocab": {
+      "@id": "https://vocabulary.lifeengine.io/Business/#",
+      "@prefix": true
+    },
+    "schema": {
+      "@id": "https://schema.org/",
+      "@prefix": true
+    },
+    "name": "rov:legalname",
+    "registration_date": "vocab:RegistrationDate",
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.jsonld:17]: https://vocabulary.lifeengine.io/Business/#RegistrationDate

image

Will not accept until https://github.com/digitalliving/vocabulary/pull/1/files is merged

**Comment by lietu** [comment]
Do a search for "Prh" before you submit this again.

File: draft/DataProducts/Company/BasicInfo.jsonld (hunk 19 above)
Comment (lietu) [reply draft/DataProducts/Company/BasicInfo.jsonld:17]: Update to https://vocabulary.lifeengine.io/Business/#registrationDate and this is solved

**Comment by lietu** [comment]
Consistency is among the most important things to do in any work like this. So far it looks like we've used `windSpeed` and one of these uses `airQualityIndex`, so we should use `businessId` and so on as well. Also there should not be multiple different ways to refer to the same thing, it's either a `companyId` or a `businessId`, or the difference between the two should be clearly explained.

**Comment by lietu** [comment]
Create the `Company/BasicInfo` specs with FastAPI, and let me know when they match.

File: draft/DataProducts/Company/BasicInfo.json [hunk 20]
```diff
@@ -1,17 +1,34 @@
...
   "info": {
-    "title": "Company Information",
-    "description": "Data Product for a company",
+    "title": "Productizer",
+    "description": "Productizer for the PRH API",
     "version": "1.0.0"
   },
   "paths": {
-    "/Company/BasicInfo": {
+    "/health": {
```
Comment (lietu) [inline draft/DataProducts/Company/BasicInfo.json:9]: Use `include_in_schema=False` for the route to remove this.

**Pull request merged**